### PR TITLE
Get rid of extra db calls in countylimits

### PR DESCRIPTION
**Pull request by fna**

### CODE REVIEW

**Comment by fna** [comment]
@amymok could you confirm that those extra calls are gone now?

File: countylimits/models.py [hunk 1]
```diff
@@ -54,12 +54,22 @@ def county_limits_by_state(state):
...
         # state value can be a State FIPS or a state abbr.
+        result = County.objects.filter(models.Q(state__state_fips=state) | models.Q(state__state_abbr=state))
+        counties = {}
+        state_abbr = ''
+        state_fips = ''
+        for county in result:
+            if not state_abbr:
+                state_abbr = county.state.state_abbr
+                state_fips = county.state.state_fips
+            counties[county.id] = [county.county_name, county.county_fips]
```
Comment (amymok) [inline countylimits/models.py]: @fna Can you use a dictionary instead to store county_name and county_fips

File: countylimits/models.py [hunk 2]
```diff
@@ -54,12 +54,22 @@ def county_limits_by_state(state):
...
+
         result = CountyLimit.objects.filter(models.Q(county__state__state_fips=state) | models.Q(county__state__state_abbr=state))
         for countylimit in result:
             data.append({
-                'state': abbr_to_name[countylimit.county.state.state_abbr],
-                'county': countylimit.county.county_name,
-                'complete_fips': '%s%s' % (countylimit.county.state.state_fips, countylimit.county.county_fips),
+                'state': abbr_to_name[state_abbr],
+                'county': counties[countylimit.county_id][0],
```
Comment (amymok) [inline countylimits/models.py]: @fna: so you don't need to use 0, can use something like "county_name"

File: countylimits/models.py [hunk 3]
```diff
@@ -54,12 +54,22 @@ def county_limits_by_state(state):
...
+                'complete_fips': '%s%s' % (state_fips, counties[countylimit.county_id][1]),
```
Comment (amymok) [inline countylimits/models.py]: @fna and 1, with "county_fips

**Comment by amymok** [comment]
@fna now is 4 queries instead of 100+! Yay!

**Comment by mandrsn** [comment]
Awesome! :+1:

**Comment by amymok** [comment]
Looks great!

**Pull request merged**